### PR TITLE
math module updated to depend on @chain-registry/utils

### PR DESCRIPTION
**Pull request by fatsheep1919**
math module updated:
1. @chain-registry/osmosis removed, functions depending on the @chain-registry/osmosis  updated to require an additional argument of assets
2. pool.ts renamed to be pool-utils.ts
3. new class LiquidityPoolCalculator added, receiving assets from constructor and pass it to the pure functions defined in pool-utils
4. @chain-registry/utils imported to invoke from utils
5. chain-registry and @chain-registry/assets imported as devDeps to run testcases

### CODE REVIEW

File: packages/math/__tests__/apr.test.ts [hunk 1]
```diff
@@ -78,18 +86,21 @@ describe("Test APR calculations", () => {
...
-        prices,
-        superfluidPools,
-        swapFee: pool.poolParams!.swapFee,
-        volume7d,
-        aprSuperfluid: 5.28,
-        includeNonPerpetual: opts.includeNonPerpetual,
-        lockup: opts.lockup,
-      });
+      const aprs = calcPoolAprs(
+        osmosisAssets,
```
Comment (pyramation) [inline packages/math/__tests__/apr.test.ts]: can I ask, why not include the property in the object below, maybe `assets`?

File: packages/math/src/apr.ts [hunk 2]
```diff
@@ -24,18 +25,21 @@ const convertLockup = (lockup: string, durations: Duration[]) => {
...
-  superfluidPools,
-  aprSuperfluid,
-  lockupDurations,
-  volume7d,
-  swapFee,
-  lockup = "14",
-  includeNonPerpetual = true,
-}: CalcPoolAprsParams) => {
+export const calcPoolAprs = (
+  osmosisAssets: OsmosisAsset[],
```
Comment (pyramation) [inline packages/math/src/apr.ts]: can we use `Asset` or `AssetList[]` from `@chain-registry/types`? or is has this been renamed?

File: packages/math/src/pool.ts [hunk 3]
```diff
@@ -75,11 +77,12 @@ export const convertDollarValueToCoins = (
 };
 
 export const convertDollarValueToShares = (
+  osmosisAssets: OsmosisAsset[],
```
Comment (pyramation) [inline packages/math/src/pool.ts]: same as above, let's use `Asset[]` or `AssetList[]` from chain-registry/types and avoid making new types

**Review by pyramation — CHANGES_REQUESTED** [review]
Great work! 

Can we also create a class like we did for ChainRegistry? 

The class would allow it so we don't have to pass the assets into every function, but would pass into the constructor. 

Maybe the class can be called `LiquidityPoolCalculator`, or something like this?

File: packages/math/src/apr.ts (hunk 2 above)
Comment (fatsheep1919) [reply packages/math/src/apr.ts]: yes we can use `Asset` directly, actually the `Asset` is just renamed to be `osmosisAsset` here.
```
import { Asset as OsmosisAsset } from "@chain-registry/types";
```

File: packages/math/src/pool.ts (hunk 3 above)
Comment (fatsheep1919) [reply packages/math/src/pool.ts]: ok no problem

File: packages/math/__tests__/apr.test.ts (hunk 1 above)
Comment (fatsheep1919) [reply packages/math/__tests__/apr.test.ts]: oh it is my oversight.
it will be better to include an assets property in the parameter object.

**Pull request merged**